R/mbbs.R

```diff
@@ -36,24 +36,38 @@ create_stop_level_counts_0 <- function(ebird, taxonomy, config = config) {
         !anyNA(df$stop_num),
         msg = "Stop data has NA values in stop_num."
       )
-
-      # Check that there are not duplicate entries
+      
+      # Check that there are not duplicate entries with different counts
       df |>
         dplyr::group_by(year, route, stop_num, common_name) |>
         dplyr::summarise(
           n = dplyr::n(),
+          distinctcounts = n_distinct(count),
           sources = paste(source, collapse = " & ")
         ) |>
-        dplyr::filter(n > 1) |>
+        dplyr::filter(n > 1 & distinctcounts > 1) |>
         (\(x) {
           logger::log_error(
             paste(
               "Route {x$route}-{x$stop_num} has multiple entries",
+              "with differing counts",
               "for {x$common_name}",
               "in {x$year} coming from {x$sources}"
             )
           )
         })()
+      
+      # Log number of duplicate entries with the same count
+      df |>
+        dplyr::distinct(year, route, stop_num, common_name, count,
+                        .keep_all = TRUE) |>
+        (\(x) {
+          logger::log_trace(
+            paste("Stop-level data contains",
+                  "{nrow(df) - nrow(x)} duplicated observations",
+                  "(same count, different sources)"))
+        })()
```

Comment:
I think I'm following the same logic here as elsewhere in the script with the loggers, but not getting anything to print out, not 100% clear to me how the df continues to be passed along within the validity checks, but adding a pipe to line 58 threw an error `Error in df : 
  The pipe operator requires a function call as RHS (<input>:21:7)`